src/coreclr/gc/gcpriv.h

```diff
@@ -4589,29 +4589,52 @@ class gc_heap
             init_recorded_tcp();
         }
 
-        bool should_change (float tcp, float* tcp_to_consider, size_t current_gc_index)
+        enum decide_change_condition
         {
+            init_change_condition = 0x0000,
+            change = 0x0001,
+            too_few_samples = 0x0002,
+            not_enough_diff_accumulated = 0x0004,
+            already_toward_target = 0x0008,
+            tcp_in_range = 0x0010,
+            temp_change = 0x0020
+        };
+
+        bool should_change (float tcp, float* tcp_to_consider, size_t current_gc_index,
```

Comment:
just a personal choice, I found it easier to look at them side by side (so have gcpriv.h in one window and gc.cpp with the callersites in a different window) since I often look at them together. it's easier than having to scroll around in one file.